src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -641,10 +655,15 @@ private void visitProperties(ObjectType objType, boolean isStatic) {
     }
 
     private void visitFunctionDeclaration(FunctionType ftype) {
-      visitFunctionType(ftype, ":");
+      visitFunctionType(ftype);
+      JSType type = ftype.getReturnType();
+      if (type != null) {
+        emit(":");
+        visitType(type);
+      }
     }
 
-    private void visitFunctionType(FunctionType ftype, String separator) {
+    private void visitFunctionType(FunctionType ftype) {
```

Comment:
I find it semi-suprising that visiting function type doesn't take care of constructors, add a comment to remind ourselves.